### ISSUE TITLE
Modal renders null when snapshotted with react-test-renderer/Jest

### ISSUE THREAD

**Issue by scherroman** [Semantic-Org/Semantic-UI-React#2454]
### Steps

LoadingModal.test.js
```
import React from 'react'
import renderer from 'react-test-renderer'
import { Modal } from 'semantic-ui-react'

test('should render correctly', () => {
    const tree = renderer.create(
        <Modal></Modal>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})
```

LoadingModal.test.js.snap
```
// Jest Snapshot v1

exports[`should render correctly 1`] = `null`;
```

### Expected Result

Modal renders normally when snapshotted with react-test-renderer and Jest

### Actual Result

Modal renders null when snapshotted with react-test-renderer and Jest

### Version
I am using react@16.2.0, jest@22.1.4, react-test-renderer@16.2.0, and semantic-ui-react@0.77.1

### Testcase
[Fork, update, and replace this pen to show the bug]:
https://codesandbox.io/s/2l3n74j9y

This doesn't look like its an issue with Jest snapshotting itself, but rather react-test-renderer with its .create() and .toJson() methods returning null.

After looking through the changelog, it looks like [this line below](https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/modules/Modal/Modal.js#L322) in Modal.js is the culprit, causing the Modal to render null when it is not rendered inside a browser. What is the purpose of short circuiting like this unless rendered inside a browser? No other components seem to do this.

```
// Short circuit when server side rendering
if (!isBrowser()) {
    return isValidElement(trigger) ? trigger : null
}
```

See [my stack overflow post](https://stackoverflow.com/questions/48373732/jest-snapshots-not-working-with-some-semantic-ui-react-components) on this issue for more context

**Comment by layershifter** [comment]
This line solved problems with SSR rendering, see #2259 for more context.

However, you can unblock this condition, see [how it done](https://github.com/Semantic-Org/Semantic-UI-React/blob/454daaab6e31459741e1cbce1b0c9a1a5f07bd2e/test/specs/lib/isBrowser-test.js#L21) in our tests.

**Comment by andrewferk** [comment]
The #2259 `isBrowser` change did not fix the issue. Even if you run the snapshot test with jsdom, `<Modal />` still renders `null`. This is because of the `<Portal />` component and the rendering happening on `document.body`.

I ended up solving this by mocking the `Portal` component.

```javascript
jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);
```

I posted more of an explanation on [stackoverflow](https://stackoverflow.com/a/49290892/651983). I'm so excited that I thought about this, because now I can use 'react-test-renderer' and `enzyme` to actually test modals now.
